app/components/session_creation_component.html.erb

```diff
@@ -1,47 +1,63 @@
-                    <%= form.submit 'Update Session', class:"btn formsubmitbtn" %>
-                <% else %>
-                    <%= form.submit 'Create Session', class:"btn formsubmitbtn" %>
-                <% end %>
-            </div>
-        <% end %>
+<turbo-frame id="session_form">
+    <% if @session.provider_session_id.present? %>
+        <h3>Edit session</h3>
+    <% else %>
+        <h3>New session</h3>
+    <% end %>
+    <div class ="card">
+        <div class="card-body">
+            <%= form_with model: @session do |form| %>
+                <div class="mb-3 ">
+                    <%= form.label :end_point, t('end_point') + ":"  %>
+                    <%= form.text_field(:end_point, placeholder:  t('end_point')) %>
+                    <% @session.errors.full_messages_for(:end_point).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
+                <div class="mb-3">
+                    <%= form.label :channel, t('channel') + ":"  %>
+                    <%= form.text_field(:channel, placeholder:  t('channel')) %>
+                    <% @session.errors.full_messages_for(:channel).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
+                <div class="mb-3">
+                    <%= form.label :topic, t('topic') + ":" %>
+                    <%= form.text_field(:topic, placeholder: t('topic')) %>
+                    <% @session.errors.full_messages_for(:topic).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
+                <div class="mb-3">
+                    <%= form.label :message_type,  t('message_type') %>
+                    <%= form.text_field(:message_type, placeholder: t('message_type')) %>
+                    <% @session.errors.full_messages_for(:message_type).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
+                <div class="mb-3">
+                    <%= form.label :user_name, t('user_name') + ":" %>
+                    <%= form.text_field(:user_name, placeholder:  t('user_name')) %>
+                    <% @session.errors.full_messages_for(:user_name).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
+                <div class="mb-3">
+                    <%= form.label :password, t('password') + ":" %>
+                    <%= form.password_field(:password, placeholder:  t('password')) %>
+                    <% @session.errors.full_messages_for(:password).each do |message| %>
+                        <div class="text-danger"><%= message %></div>
+                    <% end %>
+                </div>
```

Comment:
I noticed these are required, probably only need to be optional. Non-secure channels do not need security tokens.